### PR TITLE
boards: added support for BBC micro:bit

### PR DESCRIPTION
**Pull request by haukepetersen**
This port brings RIOT to the now freely available BBC `micro:bit`. So far it supports the on-board 5x5 LED matrix and brings SAUL mappings for the 2 buttons.

Support for the on-board sensors is missing, as the `nrf51` cpu is missing I2C support currently (but it's in the making...).

I have added a specialized test application just for this board, for testing/demonstrating the board specific LED matrix driver.

Have fun!

### CODE REVIEW

**Comment by aabadie** [comment]
I have one of these. Will try !

File: boards/microbit/microbit/microbit.c [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+#include <string.h>
+#include "xtimer.h"
+
+#include "board.h"
+#include "microbit.h"
+#include "mineplex.h"
+#include "periph/gpio.h"
+#include "periph/timer.h"
+
+#define ENABLE_DEBUG        (1)
```
Comment (kaspar030) [inline boards/microbit/microbit/microbit.c]: (0)

File: boards/microbit/include/gpio_params.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#include "saul/periph.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief    GPIO pin configuration
+ */
+static const  saul_gpio_params_t saul_gpio_params[] =
```
Comment (aabadie) [inline boards/microbit/include/gpio_params.h]: extra space between const and saul_gpio_params_t

File: boards/microbit/microbit/microbit.c (hunk 1 above)
Comment (haukepetersen) [inline boards/microbit/microbit/microbit.c]: ups

**Comment by aabadie** [comment]
Tested, works !
Next step, having  #2968 merged :)

**Comment by haukepetersen** [comment]
Yes, and then Jerryscript and we have a 'scholar' friendly platform :-)

**Comment by miri64** [comment]
ACK and go.

**Pull request merged**